abd-clam/src/utils.rs

```diff
@@ -111,3 +110,94 @@ pub fn next_ema(ratio: f64, parent_ema: f64) -> f64 {
 pub fn pos_val<T: Eq + Copy>(values: &[T], v: T) -> Option<(usize, T)> {
     values.iter().copied().enumerate().find(|&(_, x)| x == v)
 }
+
+/// Transpose a matrix represented as an array of arrays (slices) to an array of Vecs.
+///
+/// Given an array of arrays (slices), where each slice represents a row and each element
+/// within the slice represents a column, this function transposes the data to an array of Vecs.
+/// The resulting array of Vecs represents the columns of the original matrix. It is expected that each array
+/// in the input data has 6 columns.
+///
+/// # Arguments
+///
+/// - `all_ratios`: A reference to a Vec of arrays where each array has 6 columns.
+///
+/// # Returns
+///
+/// An array of Vecs where each Vec represents a column of the original matrix.
+/// /// Note that all arrays in the input Vec must have 6 columns.
+///
+/// # Panics
+///
+/// This function may panic if the input data does not have consistent array lengths or if
+/// the number of columns in the arrays is not 6.
+///
```

Comment:
This function has no panic conditions. The input is type-checked at compile time and is guaranteed to have six columns.